sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -158,6 +158,11 @@ trait HashJoin extends JoinCodegenSupport {
         output, (streamedPlan.output ++ buildPlan.output).map(_.withNullability(true)))
   }
 
+  @transient protected lazy val ignoreDuplicatedKey = joinType match {
+    case LeftExistence(_) if condition.isEmpty => true
```

Comment:
shall we make it more accurate? I think we can apply this optimization if the join condition only refers to the join keys, e.g. `t1 LEFT JOIN t2 ON t1.key = t2.key AND t2.key > 3`

Reply:
@cloud-fan - good point. Updated, and also tested it in `JoinSuite.scala`